### PR TITLE
feat: create cover page repository and hook

### PR DESCRIPTION
**Pull request by leko91**
Cover page repository and hooks for usage in navigation manager.

CC: @SamuelAlev for vis

### CODE REVIEW

File: src/AppBridgeTheme.spec.ts [hunk 1]
```diff
@@ -42,6 +61,11 @@ describe('AppBridgeThemeTest', () => {
         expect(appBridge.getProjectId()).toBe(PROJECT_ID);
     });
 
+    it('returns the hub id', () => {
+        const appBridge = new AppBridgeTheme(HUB_ID);
+        expect(appBridge.getProjectId()).toBe(HUB_ID);
```
Comment (oliverschwendener) [inline src/AppBridgeTheme.spec.ts:66]: Shouldn't this be `getHubId()`?

```suggestion
        expect(appBridge.getHubId()).toBe(HUB_ID);
```

File: src/types/CoverPage.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+export type CoverPage = {
+    id: number;
```
Comment (oliverschwendener) [inline src/types/CoverPage.ts:4]: We have to be careful with the property `id`. We don't have an entity `CoverPage` in the backend as only **one** cover page can be enabled/disabled per guideline. Therefore this property `id` could be misleading as someone could think that this is the cover page id, where in reality it is the portal id (aka guideline id).

I suggest removing it from this type, we don't use it so far right?

**Review by oliverschwendener — CHANGES_REQUESTED** [review]
See my comments.

File: src/types/CoverPage.ts (hunk 2 above)
Comment (leko91) [reply src/types/CoverPage.ts:4]: It makes sense, I just copied all the properties, I don't also see a need for an `id`

**Review by oliverschwendener — APPROVED** [review]
Code-wise this looks good to me.

Regarding the failing quality gate: either we test the remaining files or we see that this get's merged by a repository admin. @SamuelAlev

File: src/AppBridgeTheme.spec.ts [hunk 3]
```diff
@@ -42,6 +61,11 @@ describe('AppBridgeThemeTest', () => {
         expect(appBridge.getProjectId()).toBe(PROJECT_ID);
     });
 
+    it('returns the hub id', () => {
+        const appBridge = new AppBridgeTheme(HUB_ID);
```
Comment (SamuelAlev) [inline src/AppBridgeTheme.spec.ts:65]: The constructor here takes the project id

File: src/react/useCoverPage.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
```
Comment (SamuelAlev) [inline src/react/useCoverPage.ts:1]: Missing a spec file to this hook

File: src/react/useCoverPage.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+import { useEffect, useState } from 'react';
+
+import type { AppBridgeTheme } from '../AppBridgeTheme';
+import { CoverPage } from '../types';
```
Comment (SamuelAlev) [inline src/react/useCoverPage.ts:6]:
```suggestion
import type { CoverPage } from '../types';
```

File: src/AppBridgeTheme.spec.ts (hunk 3 above)
Comment (oliverschwendener) [reply src/AppBridgeTheme.spec.ts]: Done.

**Comment by oliverschwendener** [comment]
FYI: @SamuelAlev will take care of testing the react hooks later.

**Pull request merged**